test/val/val_primitives_test.cpp

```diff
@@ -28,13 +28,14 @@ using ValidatePrimitives = spvtest::ValidateBase<bool>;
 
 std::string GenerateShaderCode(
     const std::string& body,
+    const bool declare_geometry_streams_capability = true,
     const std::string& capabilities_and_extensions = "",
     const std::string& execution_model = "Geometry") {
```

Comment:
It also should validate and test that it doesn't work when execution model is not Geometry.

Validation can be done calling this function:
    _.current_function().RegisterExecutionModelLimitation(
        SpvExecutionModelGeometry,
        "Primitive instructions require Geometry execution model");
when a primitive instruction is encountered.

Reply:
(this can be done in the next pull request)

Reply:
`_.current_function().RegisterExecutionModelLimitation(
SpvExecutionModelGeometry,
"Primitive instructions require Geometry execution model");`

@atgoo 
I tried to use this naively and failed on `assert(in_function_body())` within `current_function()`.

Reply:
I've done this in the current pull request, in the last (new) commit.